vmdb/app/models/miq_ae_datastore.rb

```diff
@@ -168,10 +169,13 @@ def self.seed_default_namespace
 
   def self.reset_to_defaults
     raise "Datastore directory [#{DATASTORE_DIRECTORY}] not found" unless Dir.exist?(DATASTORE_DIRECTORY)
+    saved_attrs = preserved_attrs_for_domains
     Dir.glob(DATASTORE_DIRECTORY.join("*", MiqAeDomain::DOMAIN_YAML_FILENAME)).each do |domain_file|
       domain_name = File.basename(File.dirname(domain_file))
       reset_domain(DATASTORE_DIRECTORY, domain_name)
     end
+    saved_attrs.each { |dom, attrs| MiqAeDomain.find_by_fqname(dom).update_attributes(attrs) }
```

Comment:
I think you can use `MiqAeDomain.where(:name => dom)` instead of find_by_fqname since you know they are domains and the names are unique at that level.